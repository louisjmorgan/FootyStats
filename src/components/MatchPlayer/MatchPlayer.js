/* eslint-disable react/prop-types */
import { Container, Table } from 'react-bootstrap';

function MatchPlayer({ data }) {
  return (
    <Container fluid="xl" className="mx-auto p-2">
      <h2 className="text-center text-primary">{data.name}</h2>
      <Table className="mt-4 mx-auto w-100  mw-xl-50">
        <tbody>
          {data.stats ? Object.entries(data.stats).map(([name, value]) => {
            const normalizeName = name.replace(/([A-Z])/g, ' $1');
            const displayName = normalizeName.charAt(0).toUpperCase() + normalizeName.slice(1);
            return (
              <tr>
                <td className="text-primary ">{displayName}</td>
                <td className="text-primary">{Object.keys(value).length}</td>
              </tr>
            );
          }) : <p className="text-primary text-center">Did not play</p>}
        </tbody>
      </Table>
    </Container>
  );
}

export default MatchPlayer;
