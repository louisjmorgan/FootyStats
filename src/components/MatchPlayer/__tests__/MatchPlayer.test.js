import { render } from '@testing-library/react';
import mockMatch from '../../../mocks/responses/mockMatch.json';
import MatchPlayer from '../MatchPlayer';

const handlePlayer = (selection, match) => {
  const { players } = match[selection.isHome ? 'home' : 'away'];
  const newPlayer = players.filter((p) => p.playerId === selection.id);
  return newPlayer[0];
};

const component = (
  <MatchPlayer
    data={
      handlePlayer(
        {
          isHome: true,
          id: mockMatch.match.home.formations[0].playerIds[0],
        },
        mockMatch.match,
      )
    }
  />
);

describe('player match stats rendered correctly', () => {
  test('matches snapshot', async () => {
    const matchPlayerEl = render(component);
    expect(matchPlayerEl).toMatchSnapshot();
  });
});
