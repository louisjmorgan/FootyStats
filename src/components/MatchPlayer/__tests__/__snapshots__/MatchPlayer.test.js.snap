// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`player match stats rendered correctly matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-auto p-2 container-xl"
      >
        <h2
          class="text-center text-primary"
        >
          Rui Patrício
        </h2>
        <table
          class="mt-4 mx-auto w-100  mw-xl-50 table"
        >
          <tbody>
            <tr>
              <td
                class="text-primary "
              >
                Total Saves
              </td>
              <td
                class="text-primary"
              >
                2
              </td>
            </tr>
            <tr>
              <td
                class="text-primary "
              >
                Parried Safe
              </td>
              <td
                class="text-primary"
              >
                2
              </td>
            </tr>
            <tr>
              <td
                class="text-primary "
              >
                Claims High
              </td>
              <td
                class="text-primary"
              >
                1
              </td>
            </tr>
            <tr>
              <td
                class="text-primary "
              >
                Possession
              </td>
              <td
                class="text-primary"
              >
                15
              </td>
            </tr>
            <tr>
              <td
                class="text-primary "
              >
                Ratings
              </td>
              <td
                class="text-primary"
              >
                18
              </td>
            </tr>
            <tr>
              <td
                class="text-primary "
              >
                Touches
              </td>
              <td
                class="text-primary"
              >
                17
              </td>
            </tr>
            <tr>
              <td
                class="text-primary "
              >
                Passes Total
              </td>
              <td
                class="text-primary"
              >
                10
              </td>
            </tr>
            <tr>
              <td
                class="text-primary "
              >
                Passes Accurate
              </td>
              <td
                class="text-primary"
              >
                9
              </td>
            </tr>
            <tr>
              <td
                class="text-primary "
              >
                Pass Success
              </td>
              <td
                class="text-primary"
              >
                10
              </td>
            </tr>
            <tr>
              <td
                class="text-primary "
              >
                Aerials Total
              </td>
              <td
                class="text-primary"
              >
                1
              </td>
            </tr>
            <tr>
              <td
                class="text-primary "
              >
                Aerials Won
              </td>
              <td
                class="text-primary"
              >
                1
              </td>
            </tr>
            <tr>
              <td
                class="text-primary "
              >
                Defensive Aerials
              </td>
              <td
                class="text-primary"
              >
                1
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto p-2 container-xl"
    >
      <h2
        class="text-center text-primary"
      >
        Rui Patrício
      </h2>
      <table
        class="mt-4 mx-auto w-100  mw-xl-50 table"
      >
        <tbody>
          <tr>
            <td
              class="text-primary "
            >
              Total Saves
            </td>
            <td
              class="text-primary"
            >
              2
            </td>
          </tr>
          <tr>
            <td
              class="text-primary "
            >
              Parried Safe
            </td>
            <td
              class="text-primary"
            >
              2
            </td>
          </tr>
          <tr>
            <td
              class="text-primary "
            >
              Claims High
            </td>
            <td
              class="text-primary"
            >
              1
            </td>
          </tr>
          <tr>
            <td
              class="text-primary "
            >
              Possession
            </td>
            <td
              class="text-primary"
            >
              15
            </td>
          </tr>
          <tr>
            <td
              class="text-primary "
            >
              Ratings
            </td>
            <td
              class="text-primary"
            >
              18
            </td>
          </tr>
          <tr>
            <td
              class="text-primary "
            >
              Touches
            </td>
            <td
              class="text-primary"
            >
              17
            </td>
          </tr>
          <tr>
            <td
              class="text-primary "
            >
              Passes Total
            </td>
            <td
              class="text-primary"
            >
              10
            </td>
          </tr>
          <tr>
            <td
              class="text-primary "
            >
              Passes Accurate
            </td>
            <td
              class="text-primary"
            >
              9
            </td>
          </tr>
          <tr>
            <td
              class="text-primary "
            >
              Pass Success
            </td>
            <td
              class="text-primary"
            >
              10
            </td>
          </tr>
          <tr>
            <td
              class="text-primary "
            >
              Aerials Total
            </td>
            <td
              class="text-primary"
            >
              1
            </td>
          </tr>
          <tr>
            <td
              class="text-primary "
            >
              Aerials Won
            </td>
            <td
              class="text-primary"
            >
              1
            </td>
          </tr>
          <tr>
            <td
              class="text-primary "
            >
              Defensive Aerials
            </td>
            <td
              class="text-primary"
            >
              1
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
